### PR TITLE
 Emit more detailed type documentation for signal parameters

### LINKED ISSUE #21627: Add type information to signal documentation
**Migrated from:** https://github.com/godotengine/godot-docs/issues/1716  

At 3.1 the editor currently shows detailed type information:
> signal_types

While the documentation shows only `Object` for all non-built-in types.

Both online:
> signal_types2

And in editor:
> signal_types3

### PR DESCRIPTION
**Pull request by KellyThomas**
Fixes https://github.com/godotengine/godot/issues/21627

Thanks @neikeq

### CODE REVIEW

File: doc/classes/@GDScript.xml [hunk 1]
```diff
@@ -1177,7 +1177,7 @@
 		<constant name="INF" value="inf">
 			A positive infinity. (For negative infinity, use -INF).
 		</constant>
-		<constant name="NAN" value="nan">
+		<constant name="NAN" value="-nan(ind)">
```
Comment (akien-mga) [inline doc/classes/@GDScript.xml]: That change is weird. What system/config are you building on/for?

File: doc/classes/ProjectSettings.xml [hunk 2]
```diff
@@ -205,8 +205,6 @@
 		<member name="audio/mix_rate" type="int" setter="" getter="">
 			Mix rate used for audio. In general, it's better to not touch this and leave it to the host operating system.
 		</member>
-		<member name="audio/output_latency" type="int" setter="" getter="">
```
Comment (akien-mga) [inline doc/classes/ProjectSettings.xml]: That's weird too, all drivers seem to define it. Did you compile without audio support?

**Comment by akien-mga** [comment]
Merged manually as e00baaf6 and 29f25716 with the unexpected changes removed.

**Comment by KellyThomas** [comment]
Hmm.. I was using my usual build commands:

    call scons.bat platform=windows module_mono_enabled=yes tools=yes  mono_glue=no
    bin\godot.windows.tools.64.mono.exe --no-window --generate-mono-glue modules/mono/glue
    call scons.bat platform=windows module_mono_enabled=yes target=release_debug tools=yes

I might try a non-mono build to see if that clear sup the irregularities.

**Comment by akien-mga** [comment]
Now that Godot 3.1 has been released, we don't plan to cherry-pick new features and enhancements to the 3.0 branch, unless there is very strong support in favor of it. Removing cherrypick label for 3.0.

**Pull request closed without merging**